tests_metricflow/snapshots/test_derived_metric_rendering.py/SqlQueryPlan/DuckDB/test_cumulative_time_offset_metric_with_time_constraint__plan0_optimized.sql

```diff
@@ -42,4 +39,23 @@ FROM (
   WHERE subq_17.metric_time__day BETWEEN '2019-12-19' AND '2020-01-02'
   GROUP BY
     subq_17.metric_time__day
-) subq_23
+)
+
+, cm_5_cte AS (
```

Comment:
Is it possible to read from CTE cm_4 and compute metrics via expressions in one step? 

i.e if I we're to write this by hand I would write

cm_5_cte as (
select 
   metric_time__day
, every_2_days_bookers_2_days_ago as every_2_days_bookers_2_days_ago
from cm_4_cte
)

SELECT
 metric_time__day
 , every_2_days_bookers_2_days_ago
from cm_5_cte

Reply:
Sure, but I think following the previous comments, it seems like we need to settle on a rule for when it helps to add these CTEs.